src/shared/ui/ProgressButton.tsx

```diff
@@ -0,0 +1,44 @@
+import { FC } from 'react';
+import { ActivityIndicator, StyleProp, ViewStyle } from 'react-native';
+
+import { colors } from '@shared/lib';
+import { Text, Button } from '@shared/ui';
+
+type Color = keyof typeof colors;
+
+type Props = {
+  text: string;
+  isLoading: boolean;
+  buttonStyle?: StyleProp<ViewStyle>;
+  spinnerColor: Color;
+  variant: 'light';
+  onClick: () => void;
+};
+
+const ProgressButton: FC<Props> = ({
+  text,
+  buttonStyle,
+  onClick,
+  isLoading,
+  spinnerColor,
+  variant,
+}) => {
+  return (
+    <Button
+      variant={variant}
+      alignSelf="center"
+      onPress={onClick}
+      style={buttonStyle}
+      px={0}>
+      {isLoading ? (
+        <ActivityIndicator color={colors[spinnerColor]} />
+      ) : (
+        <Text color="$primary" fontSize={20}>
+          {text}
+        </Text>
+      )}
```

Comment:
What you can do here:
1. Instead of conditional rendering of `<Text>` you can just put opacity: 0 to make it invisible.
2. `<ActivityIndicator>` should be conditionally rendered and absolutely positioned.

That will help the component save the automatically calculated width of the Button while ActivityIndicator is showing.